src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +365,28 @@ def get_package_size(self, output):
             return Constants.UNKNOWN_PACKAGE_SIZE
     # endregion
 
+    # region auto OS updates
+    def get_current_auto_os_update_settings(self):
+        """Get current auto OS updates behaviour set on the machine using the command input"""
+        self.composite_logger.log_debug("Fetching current system settings for auto OS updates")
+        try:
+            return self.invoke_package_manager(self.find_current_auto_os_updates_settings_cmd)
+        except Exception as error:
+            error_msg = "Error fetching current system settings for auto OS updates. [Error={0}]".format(str(error))
+            self.composite_logger.log_error(error_msg)
+            raise Exception(error_msg)
+
+    def disable_auto_os_update(self):
+        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
+        try:
+            self.add_image_default_patch_mode_backup()
+            self.composite_logger.log_debug("Disabling auto OS updates if they are enabled")
+            self.invoke_package_manager(self.disable_auto_os_update_command)
```

Comment:
Manually validate that this setting has taken effect (additional code) + leave a comment why you can't validate from return code or output.